app/src/composables/tips.ts

```diff
@@ -0,0 +1,119 @@
+import { EthersJsAdapter } from '@/adapters/web3LibraryAdapter'
+import { TipsService } from '@/services/tipsService'
+import type { EventResult, TipsEventType } from '@/types'
+import dayjs from 'dayjs'
+import type { EventLog } from 'ethers'
+import { ref } from 'vue'
+
+const tipsService = new TipsService(EthersJsAdapter.getInstance())
+
+export function useTipsBalance() {
+  const balance = ref<string>('0')
+  const loading = ref(false)
+  const error = ref<any>(null)
+
+  async function getBalance() {
+    try {
+      loading.value = true
+      balance.value = await tipsService.getBalance()
+    } catch (err) {
+      error.value = err
+    } finally {
+      loading.value = false
+    }
+  }
+
+  return { balance, getBalance, loading, error }
+}
+
+export function usePushTip() {
+  const loading = ref(false)
+  const error = ref<any>(null)
+  const isSuccess = ref(false)
+
+  async function pushTip(addresses: string[], amount: number): Promise<void> {
```

Comment:
Return the transaction
Instead of returning the function directly, you return a function called execute that execute pushTip function and you can use the return value of pushTip function for the reactive value